pkg/runcfanotify/runcfanotify.go

```diff
@@ -371,7 +371,15 @@ func (n *RuncNotifier) watchPidFileIterate(pidFileDirNotify *fanotify.NotifyFD,
 	if err != nil {
 		return false, err
 	}
-	if path != pidFile {
+
+	// Some platforms (e.g. Ubuntu 22.04 on WSL2) symlink /var/run to /run.
+	// We need to evaluate the symlink before comparing with the event path below.
+	pidFileEvalSymlinks, err := filepath.EvalSymlinks(pidFile)
```

Comment:
not sure if it makes more sense to put this here or higher up when `pidFile` is first constructed. Thoughts?

Reply:
also not sure if it makes sense to also eval symlinks for `path` or not. In my testing, that seemed to use `/run` (no symlink), but maybe safer to normalize both strings being compared?